src/js/hooks/cycleCount/countStep/useCountStep.jsx

```diff
@@ -76,11 +77,18 @@ const useCountStep = () => {
     requestIdsWithDiscrepanciesRef: requestIdsWithDiscrepancies,
   });
 
-  // 5. Navigation & Header Actions
+  // 5. Validation
+  const { validateCountStep } = useCountStepValidation();
+
+  // 6. Navigation & Header Actions
   const next = useCallback(async () => {
-    await save(true);
-    setIsStepEditable(false);
-  }, [save]);
+    const validationPassed = validateCountStep();
+    dispatch(submitForm);
```

Comment:
probably I don't understand the context but why is dispatch(submitForm) called before the actual check if validation passed?
The way I would see it is either:
1. validateCountStep
2. if (validationPassed)
3. dispatch + save + setIsStepEditable

and now we call the `validateCountStep`, then we do the dispatch that doesn't care about the validate result, and then suddenly we have the if.

I would appreciate the context here